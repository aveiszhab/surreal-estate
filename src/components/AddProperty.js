import React, { useState } from "react";
import "../styles/AddProperty.css";
import { postProperty } from "../requests/requests";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      city: "Manchester",
      type: "Flat",
      bedrooms: 0,
      bathrooms: 0,
      price: 0,
      email: "",
    },
  };

  const [fields, setFields] = useState(initialState.fields);

  const handleAddProperty = (event) => {
    event.preventDefault();
    postProperty(fields);
  };

  const handleFieldChange = (event) => {
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="AddProperty">
      <form onSubmit={handleAddProperty}>
        <div className="title">
          <label className="label" htmlFor="title">
            Title:
            <input
              className="input"
              type="text"
              placeholder="Property tagline"
              id="title"
              name="title"
              value={fields.title}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <div className="city-select">
          <label className="label" htmlFor="city">
            Location:
            <select
              className="select"
              id="city"
              name="city"
              value={fields.city}
              onChange={handleFieldChange}
            >
              <option value="Manchester">Manchester</option>
              <option value="Leeds">Leeds</option>
              <option value="Sheffield">Sheffield</option>
              <option value="Liverpool">Liverpool</option>
            </select>
          </label>
        </div>

        <div className="type-select">
          <label className="label" htmlFor="type">
            Property type:
            <select
              className="select"
              id="type"
              name="type"
              value={fields.type}
              onChange={handleFieldChange}
            >
              <option value="Flat">Flat</option>
              <option value="Detached">Detached</option>
              <option value="Semi-Detached">Semi-Detached</option>
              <option value="Terraced">Terraced</option>
              <option value="End of Terrace">End of Terrace</option>
              <option value="Cottage">Cottage</option>
              <option value="Bungalow">Bungalow</option>
            </select>
          </label>
        </div>

        <div className="bedrooms">
          <label className="label" htmlFor="bedrooms">
            Number of Bedrooms:
            <input
              className="input"
              type="number"
              placeholder="Number of bedrooms"
              id="bedrooms"
              name="bedrooms"
              value={fields.bedrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <div className="bathrooms">
          <label className="label" htmlFor="bathrooms">
            Number of Bathrooms:
            <input
              className="input"
              type="number"
              placeholder="Number of bathrooms"
              id="bathrooms"
              name="bathrooms"
              value={fields.bathrooms}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <div className="price">
          <label className="label" htmlFor="price">
            Price:
            <input
              className="input"
              type="number"
              placeholder="Price of the property"
              id="price"
              name="price"
              value={fields.price}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <div className="email">
          <label className="label" htmlFor="email">
            Contact email address:
            <input
              className="input"
              type="email"
              placeholder="e.g: john.smith@email.co.uk"
              id="email"
              name="email"
              value={fields.email}
              onChange={handleFieldChange}
            />
          </label>
        </div>

        <button className="add-button" type="submit">
          Add
        </button>
      </form>
    </div>
  );
};

export default AddProperty;
