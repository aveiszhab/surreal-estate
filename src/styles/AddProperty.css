.AddProperty {
    padding:2% 25%;

}

.input,.select {
    width: 100%;
    border: 1px solid rgb(9, 32, 2);
    margin: 0.5% 1% 2% 1%;
    padding: 1%;
    box-sizing: border-box;
    font-weight: 600;
    background-color: white;
}

button[type=submit] {
    background-color: white;
    border-radius: 8px;
    border: 1px solid rgb(9, 32, 2);
    cursor: pointer;
    padding: 2% 5%;
    font-weight: 600;
    margin-left: 1%;
  }

.label{
    padding: 1%;
}

button[type=submit]:hover {
    background-color: grey;
  }

::placeholder {
    color: rgb(9, 32, 2);
}
@media only screen and (max-width: 768px) {
    [class*="input"]{
        width: 100%;    
    }
    [class*="select"]{
        width: 100%;    
    }
    [class*="AddProperty"]{
        padding:2% 10%;
    }
}
