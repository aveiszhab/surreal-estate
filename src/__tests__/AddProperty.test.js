import React from "react";
import { render } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

describe("AddProperty", () => {
  const setUp = () => {
    const { asFragment } = render(<AddProperty />);

    return {
      asFragment: asFragment(),
    };
  };

  let renderedComponent;

  beforeEach(() => {
    renderedComponent = setUp();
  });

  it("renders correctly to match the NavBar snapshot", () => {
    expect(renderedComponent.asFragment).toMatchSnapshot();
  });
});
