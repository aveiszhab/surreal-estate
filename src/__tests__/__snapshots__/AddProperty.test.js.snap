// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders correctly to match the NavBar snapshot 1`] = `
<DocumentFragment>
  <div
    class="AddProperty"
  >
    <form>
      <div
        class="title-input"
      >
        <label
          for="title"
        >
          Title:
          <input
            id="title"
            name="title"
            placeholder="Property tagline"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="city-select"
      >
        <label
          for="city"
        >
          Location:
          <select
            id="city"
            name="city"
          >
            <option
              value="Manchester"
            >
              Manchester
            </option>
            <option
              value="Leeds"
            >
              Leeds
            </option>
            <option
              value="Sheffield"
            >
              Sheffield
            </option>
            <option
              value="Liverpool"
            >
              Liverpool
            </option>
          </select>
        </label>
      </div>
      <div
        class="type-select"
      >
        <label
          for="type"
        >
          Property type:
          <select
            id="type"
            name="type"
          >
            <option
              value="Flat"
            >
              Flat
            </option>
            <option
              value="Detached"
            >
              Detached
            </option>
            <option
              value="Semi-Detached"
            >
              Semi-Detached
            </option>
            <option
              value="Terraced"
            >
              Terraced
            </option>
            <option
              value="End of Terrace"
            >
              End of Terrace
            </option>
            <option
              value="Cottage"
            >
              Cottage
            </option>
            <option
              value="Bungalow"
            >
              Bungalow
            </option>
          </select>
        </label>
      </div>
      <div
        class="bedrooms-input"
      >
        <label
          for="bedrooms"
        >
          Number of Bedrooms:
          <input
            id="bedrooms"
            name="bedrooms"
            placeholder="Number of bedrooms"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div
        class="bathrooms-input"
      >
        <label
          for="bathrooms"
        >
          Number of Bathrooms:
          <input
            id="bathrooms"
            name="bathrooms"
            placeholder="Number of bathrooms"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div
        class="price-input"
      >
        <label
          for="price"
        >
          Price:
          <input
            id="price"
            name="price"
            placeholder="Price of the property"
            type="number"
            value="0"
          />
        </label>
      </div>
      <div
        class="email-input"
      >
        <label
          for="email"
        >
          Contact email address:
          <input
            id="email"
            name="email"
            placeholder="e.g: john.smith@email.co.uk"
            type="email"
            value=""
          />
        </label>
      </div>
      <button
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
